packages/css/src/__tests__/transform.test.ts

```diff
@@ -439,4 +435,65 @@ describe('#css-transform', () => {
       `);
     });
   });
+
+  it('should add extra specificity after atomicizing without affecting class names', () => {
+    const styles = `
+      padding: 8px;
+      color: red;
+      :before {
+        content: var(--hello-world);
+        margin-right: 8px;
+        color: pink;
+      }
+    `;
+    const actual = transformCss(styles, { increaseSpecificity: true });
+    const expected = transformCss(styles, { increaseSpecificity: false });
+
+    expect(actual.classNames).toEqual(expected.classNames);
+  });
+
+  describe('increased specificity', () => {
+    it('should add extra specificity to declarations', () => {
+      const styles = `
+        padding: 8px;
+        color: red;
+        :before {
+          content: var(--hello-world);
+          margin-right: 8px;
+          color: pink;
+        }
+        ::after {
+          color: red;
+        }
+      `;
+      const { sheets: actual } = transformCss(styles, { increaseSpecificity: true });
+
+      expect(actual.join('\n')).toMatchInlineSnapshot(`
+        "._ca0qftgi:not(#\\9){padding-top:8px}
+        ._u5f3ftgi:not(#\\9){padding-right:8px}
+        ._n3tdftgi:not(#\\9){padding-bottom:8px}
+        ._19bvftgi:not(#\\9){padding-left:8px}
+        ._syaz5scu:not(#\\9){color:red}
+        ._1kt9o5oc:not(#\\9):before{content:var(--hello-world)}
+        ._eid3ftgi:not(#\\9):before{margin-right:8px}
+        ._is0632ev:not(#\\9):before{color:pink}
+        ._14rn5scu:not(#\\9):after{color:red}"
```

Comment:
I mean technically `::after` isn't `&::after` now that's properly valid modern CSS—but that's a CSS vs. Compiled spec issue, not your change, right?

Reply:
Correct, in this case it's lingering behaviour remnant of other styling libraries.